### DISCUSSION TITLE
NBT Next connection drop after 30 sec.

### DISCUSSION THREAD

**Discussion by MPi2007** [BimmerGestalt/AAIdrive#676] [Q&A]
Hi,

if i connect with usb the aaidrive apps are shown and working in idrive. After around 30 seconds all apps in idrive disapears and i can not make it shown again.
I assume that this will happen if the car has mobile reception and connect to bmw servers. If the car is in garage without mobile radio connection all apps stays in idrive.

Maybe you can do any thing to your cool app.

I have NBT-Next id4++ with installed MyBMW 2.12.3 and BMW connected classic 6.4.

**Comment by hufman** [comment]
The car's mobile reception isn't used, except for occasionally checking the ConnectedDrive subscription to enable the feature.
Does the Apps checkbox stay checked, or does it uncheck? The 30 second interval sounds like MyBMW isn't responding to the protocol heartbeat and so the connection disconnects. You should only have 1 app installed, either MyBMW or BMW Connected, and it will probably be confused about having both installed.
Another thing to check is that the MyBMW and AAIdrive apps stay running in Android: their Notifications should still be showing.

**Comment by MPi2007** [comment]
Hi Hufman,
thanks for your answer!

> Apps checkbox stay checked?
I don't have this checkbox, the apps automatic showing up under "Connected Drive > BMW Apps" menu headline.
If i install only mybmw, aaidrive is not connecting. I followed the instuction in the Q&A here (classic for the protocol and mybmw for functionality).

> MyBMW and AAIdrive apps stay running in Android
The apps are running, also in background.

Can it perhaps be due to the retrofitting of the 6nr option?

EDIT: I've tried without the classic app. The Connection works if i first use only usb, but after 30 sec all apps disappearing also.

**Reply by hufman** [reply]
Ah, you have IDrive 4, I see. MyBMW does seem to be relatively buggy with the USB connection, unfortunately. I don't believe it's due to retrofitting the option.
What functionality do you use MyBMW for? I would expect your car to not have remote connectivity anymore, and so MyBMW shouldn't be giving you any functionality and you should be able to just use BMW Connected 6.4 or Classic.

**Comment by MPi2007** [comment]
I have a working booster package subscription like remote unlock, send route to car, locate car.
I will try to test it with only the  connected 6.4 app.

**Comment by MPi2007** [comment]
I tested with deinstalled myBMW and 6.4 only and get no connection. With myBMW it is reconnecting in loop and no apps showing up.
